### ISSUE TITLE
[Bug]: Parameter offset of interface HybridSearch does not seem to take effect

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#32376]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: milvus-io-master-02f17b8-20240417
- Deployment mode(standalone or cluster):standalone
- MQ type(rocksmq, pulsar or kafka): pulsar   
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus 2.5.0rc4 and go-sdk master branch
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

1. search: limit=10 and no-offset gets ids:
```
[10812 11780 7696 10744 5859 14383 2617 6567 7759 10133]}
```
2. search: limit=5 and offset=5 gets ids:
```
[10812 11780 7696 10744 5859]}
```

- pymilvus case:
```
    @pytest.mark.tags(CaseLabel.L2)
    @pytest.mark.parametrize("offset", [5])
    @pytest.mark.parametrize("rerank", [RRFRanker()])
    @pytest.mark.parametrize("primary_field", [ct.default_int64_field_name])
    def test_hybrid_search_offset_inside_outside_params(self, primary_field, offset, rerank):
        """
        target: test hybrid search with offset inside and outside params
        method: create connection, collection, insert and search.
                Note: here the result check is through comparing the score, the ids could not be compared
                because the high probability of the same score, then the id is not fixed in the range of
                the same score
        expected: hybrid search successfully with limit(topK), and the result should be the same
        """
        # 1. initialize collection with data
        collection_w, _, _, insert_ids, time_stamp = \
            self.init_collection_general(prefix, True, primary_field=primary_field,
                                         multiple_dim_array=[default_dim, default_dim])[0:5]
        # 2. extract vector field name
        vector_name_list = cf.extract_vector_field_name_list(collection_w)
        vector_name_list.append(ct.default_float_vec_field_name)
        req_list = []
        vectors_list = []
        # 3. generate vectors
        for i in range(len(vector_name_list)):
            vectors = [[random.random() for _ in range(default_dim)] for _ in range(1)]
            vectors_list.append(vectors)
        # 5. hybrid search with offset parameter
        req_list = []
        for i in range(len(vector_name_list)):
            search_param = {
                "data": vectors_list[i],
                "anns_field": vector_name_list[i],
                "param": {"metric_type": "COSINE"},
                "limit": default_limit,
                "expr": "int64 > 0"}
            req = AnnSearchRequest(**search_param)
            req_list.append(req)
        hybrid_res = collection_w.hybrid_search(req_list, rerank, default_limit,
                                                offset=offset,
                                                check_task=CheckTasks.check_search_results,
                                                check_items={"nq": 1,
                                                             "ids": insert_ids,
                                                             "limit": default_limit})[0]
        for hits in hybrid_res:
            log.info(hits.ids)
        hybrid_res_no_offset = collection_w.hybrid_search(req_list, rerank, 5,
                                                check_task=CheckTasks.check_search_results,
                                                check_items={"nq": 1,
                                                             "ids": insert_ids,
                                                             "limit": 5})[0]
        for hits in hybrid_res_no_offset:
            log.info(hits.ids)
```
```
[2024-04-17 19:16:11,591 - INFO - ci_test]: [1383, 1459, 1700, 453, 785, 1428, 666, 1371, 1512, 185] (test_search.py:11560)
[2024-04-17 19:16:11,977 - INFO - ci_test]: [1383, 1459, 1700, 453, 785] (test_search.py:11567)
```

- go-cases:
```
func TestHybridSearchMultiVectorsPagination(t *testing.T) {
	t.Parallel()
	ctx := createContext(t, time.Second*common.DefaultTimeout*2)
	// connect
	mc := createMilvusClient(ctx, t)

	// create -> insert [0, 3000) -> flush -> index -> load
	cp := CollectionParams{CollectionFieldsType: AllVectors, AutoID: false, EnableDynamicField: false,
		ShardsNum: common.DefaultShards, Dim: common.DefaultDim}

	dp := DataParams{DoInsert: true, CollectionFieldsType: AllVectors, start: 0, nb: common.DefaultNb * 5,
		dim: common.DefaultDim, EnableDynamicField: false}

	// index params
	ips := GenDefaultIndexParamsForAllVectors()
	collName := prepareCollection(ctx, t, mc, cp, WithDataParams(dp), WithIndexParams(ips), WithCreateOption(client.WithConsistencyLevel(entity.ClStrong)))

	// hybrid search with different limit
	sp, _ := entity.NewIndexFlatSearchParam()
	expr := fmt.Sprintf("%s > 4", common.DefaultIntFieldName)
	queryVec1 := common.GenSearchVectors(1, common.DefaultDim, entity.FieldTypeFloatVector)
	queryVec2 := common.GenSearchVectors(1, common.DefaultDim, entity.FieldTypeFloat16Vector)
	// milvus ignore invalid offset with ANNSearchRequest
	for _, invalidOffset := range []int64{-20, common.MaxTopK + 1} {
		sReqs := []*client.ANNSearchRequest{
			client.NewANNSearchRequest(common.DefaultFloatVecFieldName, entity.L2, "", queryVec1, sp, common.DefaultTopK, client.WithOffset(invalidOffset)),
			client.NewANNSearchRequest(common.DefaultFloat16VecFieldName, entity.L2, "", queryVec2, sp, common.DefaultTopK),
		}
		_, errSearch := mc.HybridSearch(ctx, collName, []string{}, common.DefaultTopK, []string{}, client.NewRRFReranker(), sReqs)
		common.CheckErr(t, errSearch, true)

		//hybrid search with invalid offset
		// TODO "https://github.com/milvus-io/milvus/issues/32174"
		//_, errSearch = mc.HybridSearch(ctx, collName, []string{}, common.DefaultTopK, []string{}, client.NewRRFReranker(), sReqs, client.WithOffset(invalidOffset))
		//common.CheckErr(t, errSearch, false, "top k should be in range [1, 16384]")
	}

	// search with different reranker and offset
	for _, reranker := range []client.Reranker{
		client.NewRRFReranker(),
		client.NewWeightedReranker([]float64{0.8, 0.2}),
		client.NewWeightedReranker([]float64{0.0, 0.2}),
		client.NewWeightedReranker([]float64{0.4, 1.0}),
	} {
		sReqs := []*client.ANNSearchRequest{
			client.NewANNSearchRequest(common.DefaultFloatVecFieldName, entity.L2, expr, queryVec1, sp, common.DefaultTopK),
			client.NewANNSearchRequest(common.DefaultFloat16VecFieldName, entity.L2, expr, queryVec2, sp, common.DefaultTopK),
		}
		// hybrid search
		searchRes, errSearch := mc.HybridSearch(ctx, collName, []string{}, common.DefaultTopK, []string{}, reranker, sReqs)
		common.CheckErr(t, errSearch, true)
		offsetRes, errSearch := mc.HybridSearch(ctx, collName, []string{}, 5, []string{}, reranker, sReqs, client.WithOffset(5))
		common.CheckErr(t, errSearch, true)
		common.CheckSearchResult(t, searchRes, 1, common.DefaultTopK)
		common.CheckSearchResult(t, offsetRes, 1, 5)
		for i := 0; i < len(searchRes); i++ {
			log.Println(searchRes[i].IDs)
			log.Println(offsetRes[i].IDs)
			require.Equal(t, searchRes[i].IDs.(*entity.ColumnInt64).Data()[5:], offsetRes[i].IDs.(*entity.ColumnInt64).Data())
		}
	}
}
```

### Milvus Log

collection names: gxeV, search_collection_H4Vhazim
pods:
```
zong-go-master-2-etcd-0                                           1/1     Running                           0               20d     10.104.24.130   4am-node29   <none>           <none>
zong-go-master-2-kafka-0                                          1/1     Running                           0               20d     10.104.28.20    4am-node33   <none>           <none>
zong-go-master-2-kafka-1                                          1/1     Running                           0               20d     10.104.32.108   4am-node39   <none>           <none>
zong-go-master-2-kafka-2                                          1/1     Running                           0               20d     10.104.15.203   4am-node20   <none>           <none>
zong-go-master-2-kafka-zookeeper-0                                1/1     Running                           0               20d     10.104.28.21    4am-node33   <none>           <none>
zong-go-master-2-kafka-zookeeper-1                                1/1     Running                           0               20d     10.104.30.192   4am-node38   <none>           <none>
zong-go-master-2-kafka-zookeeper-2                                1/1     Running                           0               5d13h   10.104.23.209   4am-node27   <none>           <none>
zong-go-master-2-milvus-standalone-6b666974fc-26h8c               1/1     Running                           0               108m    10.104.29.44    4am-node35   <none>           <none>
zong-go-master-2-minio-669b58dcd8-gw7zz                           1/1     Running                           0               20d     10.104.30.193   4am-node38   <none>           <none>
```

**Comment by czs007** [comment]
This is an issue with the client. , the fix for pymilvus has already been merged. @ThreadDao 
related issue https://github.com/milvus-io/milvus/issues/32562

**Comment by ThreadDao** [comment]
@binbinlv please add related cases with pymilvus

**Comment by ThreadDao** [comment]
fixed master-20240425-2586c2f1-amd64
